src/llm/wmt_model.py

```diff
@@ -111,13 +114,11 @@ def __init__(
             f"model_dim was {model_dim} and num_heads was {num_heads}"
         )
         dim_k = dim_v = model_dim // num_heads
-        self.projection_heads = list()
+        self.projection_heads = nn.ModuleList()
```

Comment:
Thanks for opening this PR but a few of this is already addressed in #8. Can you take a look? I have not yet completed it but I can open it up for review. I am implementing the train logic there so we can tag team.

Reply:
Maybe we can merge this and rebase #8 and open up for review?

Reply:
I have made a few changes in #8. I already added the minor fixes you added here into that PR. 🥲

Reply:
Sorry I should have informed you sooner.

Reply:
How would you like me to progress here? I am open to closing this PR in favour of #8

Reply:
If the changes in 8 matches with your PR, let's close this one.

Reply:
Cool, feel free to close this PR.